### DISCUSSION TITLE
Tags

### DISCUSSION THREAD

**Discussion by Klimm** [bb-Ricardo/netbox-sync#346] [Q&A] [answered]
Hello.

I need to ask you for help.
I have VMware, virtual machines have tags.
I have installed an additional library.
But they are not synchronized with NetBox.
I specified the following parameters:
vm_tag_source = object, cluster

But I also specified all available:
vm_tag_source = object, cluster, parent_folder_1, parent_folder_2, datacenter

I'm attaching a screenshot.

I will be grateful for your help.

Thanks for the great script.
![tags](https://github.com/bb-Ricardo/netbox-sync/assets/13413364/ce8b9cc1-f3d0-4404-9c10-13f7834784f0)

**Comment by bb-Ricardo** [comment]
Hi,

Good question. This usually works.
can you send a log of an example VM where it shows the tags in vCenter and the log output of the script? The option `-l DEBUG2` would be very helpful there.

thank you

**Reply by Klimm** [reply]
For now, thanks for the reply, I found the message that /api vCenter is unavailable.
After that I got a message that tag synchronization is disabled.

Also, I apologize for my English, I am writing through a translator.

`2023-09-14 17:20:46,131 - WARNING: Unable to connect to vCenter API instance '***' on port 443: HTTPSConnectionPool(host='***', port=443): Max retries exceeded with url: /api (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)')))
2023-09-14 17:20:46,131 - WARNING: Tag syncing will be disabled.
`
I will try to download the certificate and add it to the configuration.

Thank you again.

**Reply by bb-Ricardo** [reply]
Hey,

Don't worry about your english, it's great.

Did set the option `validate_tls_certs` to true?

**Reply by Klimm** [reply]
Yes of course.  In general, synchronization works correctly, but this is what breaks everything :-)

**Reply by Klimm** [reply, marked as answer]
I managed to fix it.
I will tell you how, maybe someone will need it, also with self-signed certificates and a closed network.
I added the entire chain of my self-signed certificates to the root certificate file of the Python virtual environment.
You can find out where your environment's root certificate is located with the following command.
But don't forget to add the entire chain.
`python3 -m certifi`
I am continuing and hope that everything will be fine.

Thank you for your work.

**Reply by Klimm** [reply]
I will close this discussion if all goes well :-)
Thanks again.
